### ISSUE TITLE
Do not know how to run inference machine on .xml and .bin pair of files

### ISSUE THREAD

**Issue by jdc711** [openvinotoolkit/openvino#8371]
##### System information (version)
- OpenVINO => donwloaded from this github today
- Operating System / Platform => MACOSX
- Compiler => XCODE 
- Problem classification: Running Inference on NCS2 Stick
- Framework: IR (if applicable)

##### Detailed description
I have a .xml and .bin file that I have successfully obtained from converting the model to IR.I have donwloaded openvino from the github and have finished downloading everything and ready to run. I have also checked and the MYRIAD NCS2 device is detected afer running helloquerydevice. However, in the examples shown in this tutorial, only seems to show how to run the default samples given in the github download (such as hello_classification, hello_reshape_ssd, classification_sample_async, hello_nv12_input_classification_c) and the command given in the turotial to run the sample was: 

./object_detection_sample_ssd -i walk.bmp -m person-vehicle-bike-detection-crossroad-0078.xml async -d MYRIAD.

However, I want to be able to run my pair of .xml and .bin files on the NC2 stick. How do I do this (what command and parameters should I run?)

##### Issue submission checklist

 - [ ] I report the issue, it's not a question
   
 - [. ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ] There is reproducer code and related data files: images, videos, models, etc.

**Comment by brmarkus** [comment]
Not sure I understood you right.

You are able to run "hello_query_device" to see the MYRIAD-device (MyriadX, NCS2) being detected.

You are also able to run certain other samples, like "hello_classification, hello_reshape_ssd, classification_sample_async, hello_nv12_input_classification_c" with using "-d MYRIAD" using pre-trained models you have downloaded from Open-Model-Zoo?
Just to confirm - you can run models from OpenVINO inference-engine samples with the Myriad-X successfully?

But you can't run your model (the pair of XML&BIN file) using "-d MYRIAD" with the samples?

Your model might have special one or multiple inputs and one or multiple outputs requiring maybe special pre- and post-processing steps.

You should be able to use "benchmark_app" with "-m <path and filename of your XML-file>" and "-d MYRIAD" (even without "-i" or other parameters; default values will be chosen), as it's very generic code.

**Comment by jdc711** [comment]
Oh, I just was not sure what the general command is to run my own model. As you mentioned, is the benchmark_app what I should run?  would the command line flags be for the .xml and .bin of my model? Thanks for getting back to me.

**Comment by jdc711** [comment]
Just to follow up, do you thiink you can give me a sample command format to run my own xml and bin file?

**Comment by alalek** [comment]
Try to play with `-m person-vehicle-bike-detection-crossroad-0078.xml` parameter.
Use `--help` to see full list of options.

Note: your model must be "compatible" with demo code (follow conventions for inputs and outputs).

**Comment by jdc711** [comment]
Hi, I had a question about execution time on Neural Compute Stick 2:

I was hoping to get some inference execution speed boost by running a simple neural network model on the NC2 stick instead of CPU. However, my initial experiment showed that the performance on myriad neural compute stick 2 is MUCH  slower than CPU:

For NC2 stick:
[ INFO ] First inference took 33.88 ms
[Step 11/11] Dumping statistics report
Count:      2596 iterations
Duration:   60141.63 ms
Latency:    92.60 ms
Throughput: 5525.09 FPS

For CPU:
[ INFO ] First inference took 17.07 ms
[Step 11/11] Dumping statistics report
Count:      148124 iterations
Duration:   60001.79 ms
Latency:    1.61 ms
Throughput: 315988.43 FPS

My model is a simple linear model with input format: [128, 1, 28, 28], which gets flattened in forward to -> [128,784]:
def __init__(self):
        super(NeuralNetwork, self).__init__()
        # Layer Dim: 784 -> 512 -> 200 -> 10
        self.layers = nn.Sequential(
            nn.Linear(28*28, 512),
            nn.ReLU(),
            nn.Linear(512, 200),
            nn.ReLU(),
            nn.Linear(200, 10),
        )

    def forward(self, x):
        # Flatten x: [128, 1, 28, 28] -> [128, 784]
        x = x.view(x.shape[0], -1)
        output = self.layers(x)
        return output


Wouldyou possibly have any ideas as to why my performance on NC2 is so much slower than the CPU?

**Comment by alalek** [comment]
> NC2 is so much slower than the CPU

Which CPU is used?
- is it embedded platform with Atom CPU?
- or is it HEDT platform with Xeon CPU?

NCS2 has great "performance-per-watt" (-per-dollar) metric.

It makes sense to clarify NCS2 positioning a bit.
From developers perspective, NCS2 is useful with the following scenarios:
- as part of low power platforms (NCS2 power consumption is several watts vs 100+ watts from high-performance CPUs)
- to offload DL tasks from CPU (free CPU resources for other CPU-intensive tasks or reduce power consumption, e.g. on laptops on battery power)

**Comment by jgespino** [comment]
Closing, please re-open if additional assistance is needed.

**Comment by jdc711** [comment]
For the CPU used, I am using 2.6 GHz 6-Core Intel Core i7, for Macbook pro. This is the current execution time, throughput, etc that is attained using CPU vs MYRIAD NCS2 Stick:

MYRIAD:
[Step 11/11] Dumping statistics report
Count:      2520 iterations
Duration:   60129.34 ms
Latency:    95.21 ms
Throughput: 5364.44 FPS
CPU: 
Step 11/11] Dumping statistics report
Count:      150956 iterations
Duration:   60001.44 ms
Latency:    1.59 ms
Throughput: 322031.74 FPS

**Comment by jgespino** [comment]
Hi @jdc711 

This is the expected output. As alalek mentioned above:

>  It makes sense to clarify NCS2 positioning a bit.
From developers perspective, NCS2 is useful with the following scenarios:

> - as part of low power platforms (NCS2 power consumption is several watts vs 100+ watts from high-performance CPUs)
> - to offload DL tasks from CPU (free CPU resources for other CPU-intensive tasks or reduce power consumption, e.g. on laptops on battery power)
